subprojects/core/src/main/java/org/gradle/api/tasks/WriteProperties.java

```diff
@@ -191,30 +196,58 @@ public void setEncoding(String encoding) {
      * Returns the output file to write the properties to.
      */
     @OutputFile
+    @Optional
```

Comment:
:x: The output is already tracked by `destinationFile` this should be made internal
```suggestion
    @Internal
```